### ISSUE TITLE
[F.Rq.] "Source map" for HTML to make contributing easier

### ISSUE THREAD

**Issue by YoshiRulz** [TASVideos/tasvideos#1845]
My experience with contributing to the site, being a novice to ASP.NET, has involved a lot of grepping the codebase for tag/attribute names in order to find the source file where the thing I'm looking at originates. I've since learned that the directory structure under `/TASVideos/Pages` roughly corresponds to page paths, but that doesn't account for 'partials'.
I propose that every source file which "renders" output should also include its own path as a simple HTML comment, such that more people might be able to submit fixes or at least precisely identify where problems are.
I've found that `Html.ViewContext.ExecutingFilePath` produces the right path (relative to `$(ProjectDir)`), but it would be a pain to manually include that in every page and partial. Does anyone know of a better way?
Have a couple related SO links: [1](https://stackoverflow.com/questions/4248125/how-can-you-find-which-view-called-which-partial-view-in-asp-net-mvc), [2](https://stackoverflow.com/questions/6311463/how-to-get-the-current-view-name-in-asp-net-mvc-3)

**Comment by nattthebear** [comment]
I agree that the file structure is difficult to follow.  ASP.NET is heavily convention based and I haven't kept up on the conventions.

> I propose that every source file which 'renders' output should also include its own path as a simple HTML comment

Does that mean the final emitted markup served to the end user would have these comments in it?  Would that inflate our document size too much?

**Comment by YoshiRulz** [comment]
Just as a ballpark figure I'd say the worst-affected pages would grow by at most 1 KiB *before compression*. This could be locked to only development environments if that's a problem, given that contributors are expected to test locally anyway.

**Comment by Masterjun3** [comment]
One idea is to add one line to `_ViewStart.cshtml` that does something similar.
`<!-- @(Html.ViewContext.View.Path) -->`
Result:
![image](https://github.com/TASVideos/tasvideos/assets/22375320/1dc4786b-40c5-4e08-b69b-5ba7585f46ea)

But this [only works for views, not for partials](https://learn.microsoft.com/en-us/aspnet/core/mvc/views/layout?view=aspnetcore-8.0#running-code-before-each-view).

**Comment by Masterjun3** [comment]
Okay actually, we can override (or I think it's called to shadow) the `Activate` method of the `RazorPageActivator` and use that in the services.

![image](https://github.com/TASVideos/tasvideos/assets/22375320/d8377e88-81ef-4f87-9281-cf5248b9767d)

This way we see the beginning of every new view or partial or other things. (Not the end though, since there is no "Deactivate".)

![image](https://github.com/TASVideos/tasvideos/assets/22375320/0b304b10-1ab0-47bf-860d-be2b6e3f31d4)

**Comment by YoshiRulz** [comment]
What if you print after the `base` call?

**Comment by Masterjun3** [comment]
It's the same. The Activate call doesn't write the contents, that's why I needed a Deactivate call, which doesn't exist.
